syntax = "proto3";

package computationtocomputation;

// 引数無しにする際に使用
import "google/protobuf/empty.proto";

/**
 * ComputationToComputation service
*/
service ComputationToComputation {
    // send share to another computation container
    rpc ExchangeShares(stream Shares) returns(google.protobuf.Empty) {}
}
/**
 share_id : share_id
 job_id : job_id_number
 thread_id : thread_id_number
*/
message Address{
    int32 share_id = 1;
    uint32 job_id = 2;
    int32 thread_id = 3;
    int32 party_id = 4; // the id of party(0, 1, 2)
}

message BigIntByte{
    bool sgn = 1;
    bytes byte = 2;
}

message Share{
    oneof value{
        int64 num = 1;
        BigIntByte fp = 2;
        bool flag = 3;
    }
}

/**
 * the message of shares
*/
message Shares {
    repeated Share share_list = 1;
    Address address_id = 2;
}